stdlib/public/core/LazySequence.swift

```diff
@@ -10,124 +10,124 @@
+///                 nextPartialResult: nextPartialResult)
+///         }
 ///     }
 ///
-/// and finally, we can give all lazy sequences a lazy `scan` method:
+/// Finally, you can give all lazy sequences a lazy `scan(_:_:)` method:
 ///     
 ///     extension LazySequenceProtocol {
-///       /// Returns a sequence containing the results of
-///       ///
-///       ///   p.reduce(initial, nextPartialResult)
-///       ///
-///       /// for each prefix `p` of `self`, in order from shortest to
-///       /// longest.  For example:
-///       ///
-///       ///     Array((1..<6).lazy.scan(0, +)) // [0, 1, 3, 6, 10, 15]
-///       ///
-///       /// - Complexity: O(1)
-///       func scan<ResultElement>(
-///         _ initial: ResultElement,
-///         _ nextPartialResult: @escaping (ResultElement, Element) -> ResultElement
-///       ) -> LazyScanSequence<Self, ResultElement> {
-///         return LazyScanSequence(
-///           initial: initial, base: self, nextPartialResult: nextPartialResult)
-///       }
+///         func scan<Result>(
+///             _ initial: Result,
+///             _ nextPartialResult: @escaping (Result, Element) -> Result
+///         ) -> LazyScanSequence<Self, Result> {
+///             return LazyScanSequence(
+///                 initial: initial, base: self, nextPartialResult: nextPartialResult)
+///         }
 ///     }
 ///
-/// - See also: `LazySequence`
+/// With this type and extension method, you can call `.lazy.scan(_:_:)` on any
+/// sequence to create a lazily computed scan. The resulting `LazyScanSequence`
+/// is itself lazy, too, so further sequence operations also defer computation.
 ///
-/// - Note: The explicit permission to implement further operations
-///   lazily applies only in contexts where the sequence is statically
-///   known to conform to `LazySequenceProtocol`.  Thus, side-effects such
-///   as the accumulation of `result` below are never unexpectedly
-///   dropped or deferred:
+/// The explicit permission to implement operations lazily applies 
+/// only in contexts where the sequence is statically known to conform to
+/// `LazySequenceProtocol`. In the following example, because the extension 
+/// applies only to `Sequence`, side-effects such as the accumulation of
+/// `result` are never unexpectedly dropped or deferred:
 ///
-///       extension Sequence where Element == Int {
+///     extension Sequence where Element == Int {
 ///         func sum() -> Int {
-///           var result = 0
-///           _ = self.map { result += $0 }
-///           return result
+///             var result = 0
+///             _ = self.map { result += $0 }
+///             return result
 ///         }
-///       }
+///     }
 ///
-///   [We don't recommend that you use `map` this way, because it
-///   creates and discards an array. `sum` would be better implemented
-///   using `reduce`].
+/// Don't actually use `map` for this purpose, however, since it creates 
```

Comment:
[ASG](https://help.apple.com/applestyleguide/#/apsge70df12b) uses "since" only in the temporal sense.

```suggestion
/// Don't actually use `map` for this purpose, however, because it creates 
```